src/dotnet/Chat.Service/ChatsMigrationBackend.cs

```diff
@@ -0,0 +1,449 @@
+        updateCount = await dbContext.Reactions
+            .Where(c => c.Id.StartsWith(chatSid))
+            .ExecuteUpdateAsync(setters => setters
+                    .SetProperty(c => c.EntryId,
+                        c => c.EntryId.Replace(chatSid, newChatSid))
+                    .SetProperty(c => c.Id,
+                        c => c.Id.Replace(chatSid, newChatSid)),
+                cancellationToken)
+            .ConfigureAwait(false);
+#pragma warning restore CA1307
+
+        Log.LogInformation("Updated EntryId and Id for {Count} reaction records", updateCount);
+    }
+
+    private async Task UpdateReactionSummaries(
+        ChatDbContext dbContext,
+        string chatSid,
+        string newChatSid,
+        IReadOnlyCollection<MigratedAuthor> migratedAuthors,
+        CancellationToken cancellationToken)
+    {
+        #pragma warning disable CA1307
+        var updateCount = await dbContext.ReactionSummaries
+            .Where(c => c.Id.StartsWith(chatSid))
+            .ExecuteUpdateAsync(setters => setters
+                    .SetProperty(c => c.EntryId,
+                        c => c.EntryId.Replace(chatSid, newChatSid))
+                    .SetProperty(c => c.Id,
+                        c => c.Id.Replace(chatSid, newChatSid)),
+                cancellationToken)
+            .ConfigureAwait(false);
+#pragma warning restore CA1307
+
+        Log.LogInformation("Updated EntryId and Id for {Count} reaction summary records", updateCount);
+
+        updateCount = 0;
+        foreach (var dbSummary in dbContext.ReactionSummaries.Where(c => c.Id.StartsWith(chatSid))) {
+            var summary = dbSummary.ToModel();
+            var authorIds = summary.FirstAuthorIds;
+            var newAuthorIds = ImmutableList<AuthorId>.Empty;
+            foreach (var authorId in authorIds) {
+                var newAuthorId = migratedAuthors.First(c => OrdinalEquals(c.OriginalAuthor.Id.Value, authorId.Value)).NewId;
+                newAuthorIds = newAuthorIds.Add(newAuthorId);
+            }
+            summary = summary with { FirstAuthorIds = newAuthorIds };
+            dbSummary.UpdateFrom(summary);
+            updateCount++;
+        }
+
+        await dbContext.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
+
+        Log.LogInformation("Updated AuthorIds for {Count} reaction summary records", updateCount);
+    }
+
+    private static void AssertOneUpdated(int updateCount)
+        => AssertXUpdated(updateCount, 1);
+
+    private static void AssertXUpdated(int updateCount, int expectedUpdateCount)
+    {
+        if (updateCount != expectedUpdateCount)
+            throw new InvalidOperationException($"Expected {expectedUpdateCount} row should be update, but {updateCount} rows were updated.");
+    }
+
+    private static void AssertAtLeastOneUpdated(int updateCount)
+    {
+        if (updateCount < 1)
+            throw new InvalidOperationException("Expected at least 1 row should be update.");
```

Comment:
"Expected at least one row to be updated, though no rows were."

Reply:
It also feels like it's better to throw VersionMismatch or something like this here - i.e. any `ITransientError`.

Reply:
I would prefer not to use VersionMismatch or any other `ITransientError` since main purpose of this asserts was to ensure that written query is correct.